[Azure Search] Script install of build tools and update generate.ps1

bitsforthought opened the pull request:
I've confirmed that generate.ps1 now works for each of the SDK projects. You have to run Install-BuildTools as a prerequisite.

FYI @natinimni @ishansrivastava90 @updixit @shmed @arv100kri @weshaggard @chidozieononiwu 

If you approve this PR and you have merge permission, please merge. Thanks :)

bitsforthought committed:
[Azure Search] Script install of build tools and update generate.ps1

Commit:
[Azure Search] Fix management SDK build

brjohnstmsft commented:
FYI @shahabhijeet There are some changes in this PR for building our management SDK too.

weshaggard commented on sdk/search/Install-BuildTools.ps1:
```diff
@@ -0,0 +1,3 @@
+$scriptDir = Split-Path $MyInvocation.MyCommand.Path -Parent
```
@brjohnstmsft any reason to not directly include this in your generate.ps1 script? Do you expect folks to call it independently?

brjohnstmsft replied:
Historically everyone would just run msbuild build.proj at the root of the repo once after doing git clean -dxf, so if you never clean, you never need to run it again. The intended usage model of this is the same.

weshaggard replied:
This is only setting up the PS modules and thus is only interesting for people that call generate.ps1 (or work directly with the PS cmdlets). For that reason it fells like adding these as a step in the generate.ps1 scripts would help people not have to remember to see if this is setup correctly or not.

brjohnstmsft replied:
Sure, I can make that change.

weshaggard commented on sdk/search/Build-SearchPackages.ps1:
```diff
@@ -1,7 +1,7 @@
+dotnet msbuild "$scriptDir\..\..\mgmt.proj" /t:CreateNugetPackage /p:Scope=Search
 
 dotnet restore
 dotnet build "$engDir/service.proj" /p:ServiceDirectory="search"
```
These commands are redundant, as dotnet test or dotnet pack will also do a restore and build as a prestep. So I think all that is needed is the dotnet test and pack.

brjohnstmsft replied:
It wasn't the case when I was trying to fix this. I can try again and see if things have improved, but they were definitely broken before I added the explicit "dotnet restore". As for the other commands, I put them there on the recommendation of @chidozieononiwu.

weshaggard commented on sdk/search/Build-SearchPackages.ps1:
```diff
@@ -1,7 +1,7 @@
 $engDir = "$scriptDir\..\..\eng\"
 
-msbuild "$scriptDir\..\..\build.proj" /t:CreateNugetPackage /p:Scope="search"
+dotnet msbuild "$scriptDir\..\..\mgmt.proj" /t:CreateNugetPackage /p:Scope=Search
```
FYI - My goal is to have get the mgmt and data plane builds working similar so all you should need to do is to call "dotnet test" and "dotnet pack" on the root build.proj with the scope set to search. Currently not all the work is done to enable that but it is my eventual goal.

brjohnstmsft replied:
That would be awesome! In the meantime, if we haven't achieved this already, please ensure that all the Search libraries are building in the CI build so that build breaks are caught early (ideally before they hit master and disrupt my team).

Commit:
[Azure Search] Remove redundant commands from Build-SearchPackages

Also add Install-BuildTools.ps1 to generate.ps1 so nobody has to remember to
run it.
